python/tests/integration/connectors/memory/test_qdrant_memory_store.py

```diff
@@ -0,0 +1,246 @@
+    return MemoryRecord(
+        id="test_id1",
+        text="sample text1",
+        is_reference=False,
+        embedding=np.array([0.5, 0.5]),
+        description="description",
+        additional_metadata="additional metadata",
+        external_source_name="external source",
+        timestamp=datetime.now(),
+    )
+
+
+@pytest.fixture
+def memory_record2():
+    return MemoryRecord(
+        id="test_id2",
+        text="sample text2",
+        is_reference=False,
+        embedding=np.array([0.25, 0.75]),
+        description="description",
+        additional_metadata="additional metadata",
+        external_source_name="external source",
+        timestamp=datetime.now(),
+    )
+
+
+@pytest.fixture
+def memory_record3():
+    return MemoryRecord(
+        id="test_id3",
+        text="sample text3",
+        is_reference=False,
+        embedding=np.array([0.25, 0.80]),
+        description="description",
+        additional_metadata="additional metadata",
+        external_source_name="external source",
+        timestamp=datetime.now(),
+    )
+
+
+def test_qdrant_constructor():
+    qdrant_mem_store = QdrantMemoryStore(vector_size=TEST_VECTOR_SIZE, local=True)
+    assert qdrant_mem_store._qdrantclient is not None
+
+
+@pytest.mark.asyncio
+async def test_create_and_get_collection_async():
+    qdrant_mem_store = QdrantMemoryStore(vector_size=TEST_VECTOR_SIZE, local=True)
+
+    await qdrant_mem_store.create_collection_async("test_collection")
+    result = await qdrant_mem_store.get_collection_async("test_collection")
+    assert result.status == "green"
+
+
+@pytest.mark.asyncio
+async def test_get_collections_async():
+    qdrant_mem_store = QdrantMemoryStore(vector_size=TEST_VECTOR_SIZE, local=True)
+
+    await qdrant_mem_store.create_collection_async("test_collection1")
+    await qdrant_mem_store.create_collection_async("test_collection2")
+    await qdrant_mem_store.create_collection_async("test_collection3")
+    result = await qdrant_mem_store.get_collections_async()
+    assert len(result) == 3
+
+
+@pytest.mark.asyncio
+async def test_delete_collection_async():
+    qdrant_mem_store = QdrantMemoryStore(vector_size=TEST_VECTOR_SIZE, local=True)
+
+    await qdrant_mem_store.create_collection_async(
+        "test_collection4",
+    )
+    result = await qdrant_mem_store.get_collections_async()
+    assert len(result) == 1
+    await qdrant_mem_store.delete_collection_async("test_collection4")
+    result = await qdrant_mem_store.get_collections_async()
+    assert len(result) == 0
+
+
+@pytest.mark.asyncio
+async def test_does_collection_exist_async():
+    qdrant_mem_store = QdrantMemoryStore(vector_size=TEST_VECTOR_SIZE, local=True)
+
+    await qdrant_mem_store.create_collection_async("test_collection")
+    result = await qdrant_mem_store.does_collection_exist_async("test_collection")
+    assert result is True
+    result = await qdrant_mem_store.does_collection_exist_async("test_collection2")
```

Comment:
In test `test_get_collections_async` we are creating collection with name `test_collection2`, but not removing it. In this test we expect `False` from `does_collection_exist_async` method for collection with the same name `test_collection2`. I'm wondering, what would be the behavior if we run tests in parallel, is this a case? There is a chance that `does_collection_exist_async` will actually return `True`, so the test will fail.

We can add logic that will clean all collections before each test or use random names for test collections (e.g. using GUIDs) so the tests won't overlap. 

Please let me know if I missed something.

Reply:
We certainly could randomize collection names, but I wasn't too worried about it here because the Qdrant integration tests use a local, embedded client that does not persist between test runs, nor is it accessible outside of the test runner. 

If were to change this and point the tests to a standalone Qdrant service, I would completely agree with adding more cleanup logic.